cpp/include/Ice/ObjectAdapter.h

```diff
@@ -94,288 +87,211 @@ namespace Ice
         }
 
-        /// Add a default servant to handle requests for a specific category. Adding a default servant for a category
-        /// for which a default servant is already registered throws {@link AlreadyRegisteredException}. To dispatch
-        /// operation calls on servants, the object adapter tries to find a servant for a given Ice object identity and
-        /// facet in the following order: <ol> <li>The object adapter tries to find a servant for the identity and facet
-        /// in the Active Servant Map.</li> <li>If no servant has been found in the Active Servant Map, the object
-        /// adapter tries to find a default servant for the category component of the identity.</li> <li>If no servant
-        /// has been found by any of the preceding steps, the object adapter tries to find a default servant for an
-        /// empty category, regardless of the category contained in the identity.</li> <li>If no servant has been found
-        /// by any of the preceding steps, the object adapter gives up and the caller receives {@link
-        /// ObjectNotExistException} or {@link FacetNotExistException}.</li>
-        /// </ol>
-        /// @param servant The default servant.
-        /// @param category The category for which the default servant is registered. An empty category means it will
-        /// handle all categories.
-        /// @see #removeDefaultServant
-        /// @see #findDefaultServant
+        /// Adds a default servant to handle requests for a specific category. When an object adapter dispatches an
+        /// incoming request, it tries to find a servant for the identity and facet carried by the request in the
+        /// following order:
+        ///  - The object adapter tries to find a servant for the identity and facet in the Active Servant Map.
+        ///  - If this fails, the object adapter tries to find a default servant for the category component of the
+        ///    identity.
+        ///  - If this fails, the object adapter tries to find a default servant for the empty category, regardless of
+        ///    the category contained in the identity.
+        ///  - If this fails, the object adapter tries to find a servant locator for the category component of the
+        ///    identity. If there is no such servant locator, the object adapter tries to find a servant locator for the
+        ///    empty category.
+        ///    - If a servant locator is found, the object adapter tries to find a servant using this servant locator.
+        ///  - If all the previous steps fail, the object adapter gives up and the caller receives an
+        ///    ObjectNotExistException or a FacetNotExistException.
+        /// @param servant The default servant to add.
+        /// @param category The category for which the default servant is registered. The empty category means it
+        /// handles all categories.
```

Comment:
What about mentioning collocated transport would not consider default servants when checking if an object is local.

Reply:
It's actually more complicated and hard to explain succinctly.

See https://github.com/zeroc-ice/ice/blob/80c9e6baac79b168a25f5193a5175359cedc37de/cpp/src/Ice/ObjectAdapterI.cpp#L650